ambry-account/src/main/java/com/github/ambry/account/CompositeAccountService.java

```diff
@@ -113,79 +107,81 @@ public boolean removeAccountUpdateConsumer(Consumer<Collection<Account>> account
-    Set<Container> secondaryResult = secondaryAccountService.getContainersByStatus(containerStatus);
-    if (!primaryResult.equals(secondaryResult)) {
-      logger.warn(
-          "Inconsistency detected between primary and secondary for containers with status ={}, primary ={}, secondary = {}",
-          containerStatus, primaryResult, secondaryResult);
-      accountServiceMetrics.getContainerDataInconsistencyCount.inc();
+    if (shouldCompare()) {
+      Set<Container> secondaryResult = secondaryAccountService.getContainersByStatus(containerStatus);
+      if (!primaryResult.equals(secondaryResult)) {
+        logger.warn(
+            "Inconsistency detected between primary and secondary for containers with status ={}, primary ={}, secondary = {}",
+            containerStatus, primaryResult, secondaryResult);
+        accountServiceMetrics.accountDataInconsistencyCount.inc();
+      }
     }
     return primaryResult;
   }
 
   @Override
   public void selectInactiveContainersAndMarkInZK(StatsSnapshot statsSnapshot) {
     primaryAccountService.selectInactiveContainersAndMarkInZK(statsSnapshot);
-    try {
-      secondaryAccountService.selectInactiveContainersAndMarkInZK(statsSnapshot);
-    } catch (Exception e) {
-      logger.warn("Marking containers as inactive failed for secondary", e);
-      accountServiceMetrics.updateContainerInconsistencyCount.inc();
-    }
+    secondaryAccountService.selectInactiveContainersAndMarkInZK(statsSnapshot);
   }
 
   @Override
   public void close() throws IOException {
+    shutDownExecutorService(scheduler, config.consistencyCheckerShutdownTimeoutMinutes, TimeUnit.MINUTES);
     try {
       primaryAccountService.close();
-    } finally {
+    } catch (Exception e) {
+      logger.error("Close failed for primary source", e);
+    }
+    try {
       secondaryAccountService.close();
+    } catch (Exception e) {
+      logger.error("Close failed for secondary source", e);
     }
-    shutDownExecutorService(scheduler, config.consistencyCheckerShutdownTimeoutMinutes, TimeUnit.MINUTES);
   }
 
   /**
    * Compares and logs differences (if any) in Account metadata stored in primary and secondary sources
    */
-  private void compareAccountMetadata() {
-    Set<Account> primaryAccounts = new HashSet<>(primaryAccountService.getAllAccounts());
-    Set<Account> secondaryAccounts = new HashSet<>(secondaryAccountService.getAllAccounts());
-    if (!primaryAccounts.equals(secondaryAccounts)) {
-      accountServiceMetrics.accountMetadataInconsistencyCount.inc();
-      logger.debug("Inconsistency detected between primary and secondary at {} for accounts ={}",
-          SystemTime.getInstance().milliseconds(), primaryAccounts.removeAll(secondaryAccounts));
+  void compareAccountMetadata() {
+    if (!AccountUtils.compareAccounts(primaryAccountService.getAllAccounts(),
+        secondaryAccountService.getAllAccounts())) {
+      accountServiceMetrics.accountDataInconsistencyCount.inc();
```

Comment:
It could be useful to maintain a Guage metric of the running total of inconsistencies, so we can alert when it stays above 0 for some time threshold.  We can add this in a follow up PR.